### PR TITLE
feat(barchart): add entry option to customize a bar indicator

### PR DESCRIPTION
**Pull request by pedromonad**
[SDE-1142](https://toptal-core.atlassian.net/browse/SDE-1142)

### Description

This PR adds the `dataItem` option to `renderBarIndicators` and `renderBarIndicators`, allowing us to customize bar indicators as needed.

### How to test

- Run the storybook locally (`yarn start`)
- Navigate to the `BarChart` component (http://localhost:9001/?path=/story/picasso-charts-barchart--barchart#stacked-with-bar-chart-indicators) and find the example provided

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| image | image |

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Make sure that additions and changes on design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests

**Breaking change**

- [x] codemod is created and showcased in the changeset
- [x] test alpha package of Picasso in StaffPortal

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[SDE-1142]: https://toptal-core.atlassian.net/browse/SDE-1142?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by pedromonad** [comment]
@toptal-bot run package:alpha-release

**Comment by pedromonad** [comment]
@toptal-bot run package:alpha-release

**Comment by pedromonad** [comment]
@toptal-anvil ping reviewers

**Comment by pedromonad** [comment]
@toptal-anvil ping reviewers

File: .changeset/blue-pugs-sleep.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+'@toptal/picasso-charts': patch
```
Comment (sashuk) [inline .changeset/blue-pugs-sleep.md:2]: Is it more like a minor update, as the new feature was added?

File: packages/picasso-charts/src/BarChart/story/BarIndicator.example.tsx [hunk 2]
```diff
@@ -27,20 +26,21 @@ const CHART_DATA = [
...
 
 const Example = () => (
   <div style={{ width: 720 }}>
     <BarChart
       width='100%'
-      data={CHART_DATA}
+      data={CHART_DATA as any}
       getBarColor={() => palette.blue.main}
-      renderBarIndicators={({ dataKey }: BarOptions) => {
+      renderBarIndicators={({ dataKey, entry }) => {
```
Comment (sashuk) [inline packages/picasso-charts/src/BarChart/story/BarIndicator.example.tsx:39]: NIT: Subjectively, `entry` sounds is a bit ambiguous, how about `dataEntry` or `dataItem`?

Comment (pedromonad) [reply packages/picasso-charts/src/BarChart/story/BarIndicator.example.tsx:39]: Done!

File: .changeset/blue-pugs-sleep.md (hunk 1 above)
Comment (pedromonad) [reply .changeset/blue-pugs-sleep.md]: Updated!

**Comment by pedromonad** [comment]
@toptal-anvil ping reviewers

**Comment by sashuk** [comment]
@toptal-anvil ping reviewers

**Comment by pedromonad** [comment]
@toptal-anvil ping reviewers

**Review by mkrl — APPROVED** [review]

**Comment by pedromonad** [comment]
Could you please review Happo?

cc: @mkrl @sashuk

**Comment by sashuk** [comment]
@pedromonad Checking!

**Pull request merged**